workflow/rules/05_deleterious_dfe.smk

```diff
@@ -0,0 +1,222 @@
+    conda:
+        "../envs/env.yaml"
+    params:
+        ubounds=get_dfe_ubounds,
+        lbounds=get_dfe_lbounds,
+        ratio=get_ratio,
+        output_prefix=get_dadi_bestfits_prefix,
+        optimizations=get_optimizations,
+    resources:
+        time=7200,
+        cpus=64,
+    log:
+        "logs/deleterious_dfe/infer_dfe_fine_tune.{species}.{ppl}.{demog}.{dfe}.log",
+    shell:
+        """
+        dadi-cli InferDFE --fs {input.fs} --cache1d {input.cache} --demo-popt {input.dm_bestfit} --output-prefix {params.output_prefix} --pdf1d {wildcards.dfe} --bestfit-p0-file {input.p0} --ubounds {params.ubounds} --lbounds {params.lbounds} --ratio {params.ratio} --cpus {resources.cpus} --force-convergence {params.optimizations} 2> {log}
+        """
+
+
+rule godambe_ci:
+    input:
+        syn_vcf="results/processed_data/{species}/1pop/{ppl}/{ppl}.biallelic.syn.snps.vcf.gz",
+        nonsyn_vcf="results/processed_data/{species}/1pop/{ppl}/{ppl}.biallelic.nonsyn.snps.vcf.gz",
+        pop_info=rules.create_pop_info.output.pop_info,
+        nonsyn_fs=rules.generate_1pop_fs.output.nonsyn_fs,
+        cache=rules.generate_1d_cache.output.cache,
+        dfe_bestfit=rules.infer_dfe_fine_tune.output.bestfit,
+    output:
+        dfe_godambe_ci="results/dadi/{species}/dfe/{ppl}/StatDFE/{ppl}.{demog}.{dfe}.godambe.ci",
+    conda:
+        "../envs/env.yaml"
+    params:
+        syn_dir="results/dadi/{species}/dfe/{ppl}/StatDFE/{ppl}_bootstrapping_syn",
+        nonsyn_dir="results/dadi/{species}/dfe/{ppl}/StatDFE/{ppl}_bootstrapping_non",
+        syn_output_prefix="results/dadi/{species}/dfe/{ppl}/StatDFE/{ppl}_bootstrapping_syn/{ppl}.syn.unfolded",
+        nonsyn_output_prefix="results/dadi/{species}/dfe/{ppl}/StatDFE/{ppl}_bootstrapping_non/{ppl}.nonsyn.unfolded",
+        bootstrap_reps=bootstrap_reps,
+        chunk_size=chunk_size,
+    log:
+        "logs/deleterious_dfe/godambe_ci.{species}.{ppl}.{demog}.{dfe}.log",
+    shell:
+        """
+        [ -d {params.syn_dir} ] || mkdir {params.syn_dir}
+        [ -d {params.nonsyn_dir} ] || mkdir {params.nonsyn_dir}
+        dadi-cli GenerateFs --vcf {input.syn_vcf} --pop-info {input.pop_info} --pop-ids {wildcards.ppl} --projections $(awk 'END {{print NR * 2}}' {input.pop_info}) --polarized --bootstrap {params.bootstrap_reps} --chunk-size {params.chunk_size} --output {params.syn_output_prefix} 2> {log}
+        dadi-cli GenerateFs --vcf {input.nonsyn_vcf} --pop-info {input.pop_info} --pop-ids {wildcards.ppl} --projections $(awk 'END {{print NR * 2}}' {input.pop_info}) --polarized --bootstrap {params.bootstrap_reps} --chunk-size {params.chunk_size} --output {params.nonsyn_output_prefix} 2>> {log}
+        dadi-cli StatDFE --fs {input.nonsyn_fs} --dfe-popt {input.dfe_bestfit} --cache1d {input.cache} --pdf1d {wildcards.dfe} --bootstrapping-nonsynonymous-dir {params.syn_dir} --bootstrapping-synonymous-dir {params.nonsyn_dir} --output {output.dfe_godambe_ci} 2>> {log}
```

Comment:
Here, I made a mistake, we should use 

```
--bootstrapping-nonsynonymous-dir {params.nonsyn_dir} --bootstrapping-synonymous-dir {params.syn_dir}
```

Reply:
and the jdfe smk also needs the godambe rule, you can refer to https://github.com/admixVIE/gas/blob/main/workflow/rules/run_dadi_jdfe.smk